LGButton/Classes/LGButton.swift

```diff
@@ -513,43 +528,31 @@ public class LGButton: UIControl {
     }
     
     fileprivate func setupIcon(icon:UILabel, fontName:String, iconName:String, fontSize:CGFloat, color:UIColor){
-        icon.isHidden = !availableFontIcons.contains(fontName)
-        if  !icon.isHidden {
+        defer {
+            setupBorderAndCorners()
+        }
+        if let iconFont = SwiftIconFont.fonts[fontName] {
+            icon.isHidden = false
             icon.textColor = color
-            switch fontName {
-            case "fa":
-                icon.font = UIFont.icon(from: .FontAwesome, ofSize: fontSize)
-                icon.text = String.fontAwesomeIcon(iconName)
-                break;
-            case "io":
-                icon.font = UIFont.icon(from: .Ionicon, ofSize: fontSize)
-                icon.text = String.fontIonIcon(iconName)
-                break;
-            case "oc":
-                icon.font = UIFont.icon(from: .Octicon, ofSize: fontSize)
-                icon.text = String.fontOcticon(iconName)
-                break;
-            case "ic":
-                icon.font = UIFont.icon(from: .Iconic, ofSize: fontSize)
-                icon.text = String.fontIconicIcon(iconName)
-                break;
-            case "ma":
-                icon.font = UIFont.icon(from: .MaterialIcon, ofSize: fontSize)
-                icon.text = String.fontMaterialIcon(iconName.replacingOccurrences(of: "-", with: "."))
-                break;
-            case "ti":
-                icon.font = UIFont.icon(from: .Themify, ofSize: fontSize)
-                icon.text = String.fontThemifyIcon(iconName.replacingOccurrences(of: "-", with: "."))
-                break;
-            case "mi":
-                icon.font = UIFont.icon(from: .MapIcon, ofSize: fontSize)
-                icon.text = String.fontMapIcon(iconName.replacingOccurrences(of: "-", with: "."))
-                break;
-            default:
-                break;
+            icon.font = UIFont.icon(from: iconFont, ofSize: fontSize)
+            if let iconStr = String.getIcon(from: iconFont, code: iconName) {
+                icon.text = iconStr
+            }else{
+                let joind = ["-", ".", "_"]
+                for left in joind {
+                    for right in joind {
+                        if left == right { continue }
+                        let code = left.replacingOccurrences(of: left, with: right)
```

Comment:
this is won't return anything. it should be:

```suggestion
                        let code = iconName.replacingOccurrences(of: left, with: right)
```